http_requests.py

```diff
@@ -106,7 +106,11 @@ def _request(self,
             headers.update(kwargs['headers'])
             del kwargs['headers']
         if 'data' in kwargs:
-            headers['content-type'] = 'application/x-yaml'
```

Comment:
oh dear.. this was somewhat dirty (defaulting to x-yaml mime type). at any rate: would it not be better to simply replace the 'content-type' in case it is given via kwargs, rather than introducing this new special logic?